From: Chengming Zhou <chengming.zhou@linux.dev>
Subject: mm-zswap-use-only-one-pool-in-zswap-v2
Date: Fri, 21 Jun 2024 15:15:10 +0800

fix error handling in zswap_pool_create(), per Dan Carpenter

Link: https://lkml.kernel.org/r/20240621-zsmalloc-lock-mm-everything-v2-2-d30e9cd2b793@linux.dev
Signed-off-by: Chengming Zhou <chengming.zhou@linux.dev>
Reviewed-by: Nhat Pham <nphamcs@gmail.com>
Acked-by: Yosry Ahmed <yosryahmed@google.com>
Cc: Chengming Zhou <zhouchengming@bytedance.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Minchan Kim <minchan@kernel.org>
Cc: Sergey Senozhatsky <senozhatsky@chromium.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/zswap.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/zswap.c~mm-zswap-use-only-one-pool-in-zswap-v2
+++ a/mm/zswap.c
@@ -268,7 +268,7 @@ static struct zswap_pool *zswap_pool_cre
 	pool->zpool = zpool_create_pool(type, name, gfp);
 	if (!pool->zpool) {
 		pr_err("%s zpool not available\n", type);
-		goto error;
+		return NULL;
 	}
 	pr_debug("using %s zpool\n", zpool_get_type(pool->zpool));
 
_
